### PR TITLE
[16.0][ADD] delivery_automatic_package

### PR DESCRIPTION
**Pull request by rousseldenis**

### CODE REVIEW

File: delivery_package_automatic/models/res_company.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class ResCompany(models.Model):
+
+    _inherit = "res.company"
+
+    automatic_package_creation_default = fields.Boolean(
```
Comment (lmignon) [inline delivery_package_automatic/models/res_company.py]: s/automatic_package_creation_default/automatic_package_creation_at_delivery_default

File: delivery_package_automatic/models/res_config_settings.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class ResConfigSettings(models.TransientModel):
+
+    _inherit = "res.config.settings"
+
+    automatic_package_creation_default = fields.Boolean(
```
Comment (lmignon) [inline delivery_package_automatic/models/res_config_settings.py]: s/automatic_package_creation_default/automatic_package_creation_at_delivery_default

**Review by lmignon — CHANGES_REQUESTED** [review]
LGTM but I would use a field name that better reflect the context where it's used.

File: delivery_package_automatic/models/res_company.py (hunk 1 above)
Comment (rousseldenis) [reply delivery_package_automatic/models/res_company.py]: Done

File: delivery_package_automatic/models/res_config_settings.py (hunk 2 above)
Comment (rousseldenis) [reply delivery_package_automatic/models/res_config_settings.py]: Done

File: delivery_automatic_package/models/delivery_carrier.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class DeliveryCarrier(models.Model):
+
+    _inherit = "delivery.carrier"
+
+    automatic_package_creation_at_delivery = fields.Boolean(
```
Comment (bealdav) [inline delivery_automatic_package/models/delivery_carrier.py:11]: Nitpicking
Python code makes things automatic, then `automatic`  have no high value here and make field name longer useless.
```suggestion
    package_creation_at_delivery = fields.Boolean(
```

Comment (rousseldenis) [reply delivery_automatic_package/models/delivery_carrier.py:11]: Yes, but that reflects a change from standard behavior.

Apart that, are you ok with the change ?

**Review by lmignon — APPROVED** [review]
LGTM (Code review)

**Review by bealdav — APPROVED** [review]
Thanks a lot for this contribution, LGTM code review.

My own preference is to have only meaningful words in fields to make it shorter, but that's just my taste.

Then that's an unconditionally approve

**Review by florian-dacosta — APPROVED** [review]
Thanks a lot for this PR, I think it is a very good thing to make this feature in a separate module so it becomes optional.
LGTM code review

**Comment by bguillot** [comment]
Thanks @rousseldenis !

**Pull request merged**